### ISSUE TITLE
Actual name and wiki entry

### ISSUE THREAD

**Issue by ShalokShalom** [therecipe/qt#382]
The actual name of this binding is simply "qt"? I am asking since i write the entry in Qts own Wiki for it. 
I am wondering its not there yet..

**Comment by therecipe** [comment]
Hey

>The actual name of this binding is simply "qt"? 

Yeah, actually it's nameless.
I wasn't able to come up with something I liked and the obvious ones like "goqt" or "qtgo" were already taken, so I just called the repo "qt" and moved on. So yeah "qt" is the official name, I guess.

> I am asking since i write the entry in Qts own Wiki for it.
I am wondering its not there yet..

I'm not sure, I did know that Qt had a wiki but I thought that it was inactive.
However, I would be happy if this binding would be listed there :)

**Comment by ShalokShalom** [comment]
I think it **IS** inactive. ;) 

I add qt to this list, no acceptance yet, while Ruby and Perl, both with Qt1 until Qt4 support are still in and two other bindings are also added. https://wiki.qt.io/Language_Bindings

Maybe a ping an admin later, no reaction on Qt either yet.

**Comment by ShalokShalom** [comment]
CuteGo

**Comment by joeblew99** [comment]
Nice name :) needs some name that sticks in people's minds

**Comment by masiulaniec** [comment]
Please don't invent cute names. The ruby world is infested with project names such as "cucumber". One of the benefits of go's hierarchical url-based import path approach is that package names can be simply self-describing. In other words, workarounds from systems using flat namespaces don't apply here.

**Comment by masiulaniec** [comment]
A related question is whether the current split into large `core`, `gui`, `widgets`, etc., packages is the best possible one. It seems like if all I need is `QLabel`, then I should be able to write:
```
import "github.com/therecipe/qt/QLabel"
```
... and use just the `*QLabel.W` type ('W' for Widget), thus avoiding the cost of compiling 95% of stuff I don't need.

**Comment by therecipe** [comment]
@masiulaniec 

>A related question is whether the current split into large core, gui, widgets, etc., packages is the best possible one.
>I wonder if this approach was considered but rejected.

Yeah, I recall thinking about that when I split the large "qt" pkg from the early days into the modules.
(And also again when the "widgets" module, took 8gb ram to compile)

I think I tried and rejected it because there was a) a lot of duplicate C and Go code necessary to get it working and b) I had problems with circle dependencies.

**Comment by ShalokShalom** [comment]
@therecipe Any suggestions? I added your project now to the Qt Wiki: 
https://wiki.qt.io/Language_Bindings

**Comment by therecipe** [comment]
@ShalokShalom 
Ah, thank you :)

>Any suggestions?

Maybe it should be mentioned that you can subclass with the help of qtmoc and thereby extend your "classes" (Go has only structs and no real classes) with custom signals/slots/properties/constructors.
That rcc and {win|mac}deployqt are wrapped by qtrcc and qtdeploy, and that you can simply cross compile to most other targets (>= 10) with the help of docker and vagrant.
And that you can use docker and vagrant for your CI and CD purposes as well.
But there are also a few other things, like the fact that you can re-generate the whole binding with qtsetup against a range of Qt versions. Or that the binding is a one to one mapping against Qt's C++ api. Or maybe even that you don't have to install Qt on your pc at all, if you run the minimal setup with docker. Or that you can override virtual functions (and also signals/slots) at runtime with anonymous functions (Lambdas). And there are probably a few other notable things as well.

But I just took another look to see what has been written for the other bindings, and that was not that much :D
So I'm not sure if you can use my suggestions from above at all.

The only thing that comes close to PyQt's and PySide's docs, tutorials and such.
Is probably the slack channel: https://gophers.slack.com/messages/qt-binding/details
But even I just visit it infrequently.


Also thanks for the name suggestions.
But I'm not really sure if I want to attach a "Go" suffix or prefix at all to it.

>Is there a specific list?
Yes, you can find one here: https://github.com/therecipe/qt/blob/8ca9dc8f59110a659dabdef0201eaeb74f49efc1/internal/binding/parser/helper.go#L257-L311

And another one would be the list of folders at the root of the repo: https://github.com/therecipe/qt
Each folder (beside "cmd" and "internal") is a Qt module.

**Comment by ShalokShalom** [comment]
Thanks a lot, that helps me further.

**Comment by RadhiFadlillah** [comment]
Well, since `qt` is pronounced `kyu-tee`, what about :

- `gokyu`
- `goqu`
- `goku`

@masiulaniec I think it's fine if package's name is not self-describing, simply because number of vocabulary is limited while number of libraries or packages will keep growing. Even in Go community there are many packages whose name doesn't describing himself, like `gin`, `martini`, `cockroach`, `hugo`, etc.

**Comment by ShalokShalom** [comment]
This repo suggests to name it therecipe-qt: https://github.com/wzwmzm/therecipe-qt/commits/master ;)

**Comment by therecipe** [comment]
@RadhiFadlillah 
Also thanks for the name suggestions :)
But I'm not really sure, I'm not really a fan of "go" prefixes or suffixes to be honest.

**Comment by Huholoman** [comment]
Goqu is already taken by sql builder.

**Comment by ShalokShalom** [comment]
It might be nice to offer a template for the Qt Creator:

![screenshot_20170831_125104](https://user-images.githubusercontent.com/6344099/29919844-38410b18-8e4b-11e7-8606-288dcfcc35f1.png)

**Comment by therecipe** [comment]
@ShalokShalom Yepp that would be nice, and I think it's also already in the works: https://github.com/gotschmarcel/qtcreator-go
But I haven't really looked into that yet.

**Comment by mewmew** [comment]
If it has to be renamed, I'd go for `cute`; as that is both memorable and roughly how I could pronounce qt.

**Comment by justinclift** [comment]
@therecipe Since you're stuck when trying to decide on the name, why don't we just do a vote thing and be done with it?  As long as the end result isn't horrible that'd "get it done" so there's one less thing taking up potential thinking time. :smile:

**Comment by ShalokShalom** [comment]
Go + Qt = Goat ^-^

**Comment by amitdo** [comment]
* QtAndGo
* QG
* QGo
* Go2Qt
* Go4Qt
* Qpher (Qt+Gopher)
* Qter

**Comment by ShalokShalom** [comment]
Qopher sounds nice too

**Comment by larvasapiens** [comment]
I second "Qopher". "Quza" is another option (refering to how Mexicans refer to Gophers, "Tuza").

**Comment by tobimensch** [comment]
qtee (pronounced cutie)

**Comment by paboum** [comment]
- widGots (widgets + Go)
- cuteWidGots
- (Qt) widGet toOlkit (could be mixed in a logo)
- Quasar TechnolOGies (from the original library name ethymology)
- QGo
- QGopher
- QB4G or GB4Go (Qt Bindings for Go)
- QW4G or GW4Go (Qt Wrappers for Go)
- Goatee ("Go" + "cutie"), a style of facial hair incorporating hair on a man's chin but not his cheeks

**Comment by ilovesusu** [comment]
GoQt
like
pyqt

**Comment by ShalokShalom** [comment]
GoQt is already in use

**Comment by uvatbc** [comment]
QGoatee

**Comment by colourdelete** [comment]
goside like pyside

**Comment by tox2ik** [comment]
CuteWalk ot QuteWalk

**Comment by brenthuisman** [comment]
Qt for Go (similar to the official Python bindings).

**Comment by w3bb** [comment]
gute, it's how you pronounce qt but with a g. Or maybe just gt.
